src/linked_hash_map.rs

```diff
@@ -297,6 +297,35 @@ impl<K: Hash<HmHasher> + Eq, V> LinkedHashMap<K, V> {
         }
     }
 
+    /// A double-ended iterator visiting all key-value pairs in order of insertion.
+    /// Iterator element type is `(&'a K, &'a mut V)`
+    /// # Examples
+    /// ```rust
+    /// use collect::LinkedHashMap;
+    ///
+    /// let mut map = LinkedHashMap::new();
+    /// map.insert("a", 10);
+    /// map.insert("c", 30);
+    /// map.insert("b", 20);
+    ///
+    /// {
+    ///     let mut iter = map.iter_mut();
+    ///     let mut entry = iter.next().unwrap();
+    ///     assert_eq!(&"a", entry.0);
+    ///     *entry.1 = 17;
+    /// }
+    ///
+    /// assert_eq!(&17, map.get(&"a").unwrap());
+    /// ```
+    pub fn iter_mut<'a>(&'a mut self) -> IterMut<'a, K, V> {
```

Comment:
The `IterMut` struct declaration needs it, but the method signature can be:
```rust
fn iter_mut(&mut self) -> IterMut<K, V>
```

On Sunday, January 11, 2015, Alexis Beingessner <notifications@github.com>
wrote:

> In src/linked_hash_map.rs
> <https://github.com/Gankro/collect-rs/pull/80#discussion-diff-22769219>:
>
> > +    ///
> > +    /// let mut map = LinkedHashMap::new();
> > +    /// map.insert("a", 10);
> > +    /// map.insert("c", 30);
> > +    /// map.insert("b", 20);
> > +    ///
> > +    /// {
> > +    ///     let mut iter = map.iter_mut();
> > +    ///     let mut entry = iter.next().unwrap();
> > +    ///     assert_eq!(&"a", entry.0);
> > +    ///     *entry.1 = 17;
> > +    /// }
> > +    ///
> > +    /// assert_eq!(&17, map.get(&"a").unwrap());
> > +    /// ```
> > +    pub fn iter_mut<'a>(&'a mut self) -> IterMut<'a, K, V> {
>
> Can it? It needs to be used in the impl, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Gankro/collect-rs/pull/80/files#r22769219>.
>